packages/blocks/shuffle/src/shuffle.tsx

```diff
@@ -39,113 +41,111 @@ export const Shuffle: BlockComponent<BlockEntityProperties> = ({
   const blockRootRef = useRef<HTMLDivElement>(null);
   const { graphService } = useGraphBlockService(blockRootRef);
 
-  const [list, setList] = useState(
+  const [draftItems, setDraftItems] = useState(() =>
     createItems(items?.length ? items : initialItems),
   );
 
-  useEffect(() => {
-    if (
-      items &&
-      !isEqual(
-        items,
-        list.map((item) => item.value),
-      )
-    ) {
-      setList(createItems(items));
+  const [prevItems, setPrevItems] = useState(items);
+
+  if (items && !isEqual(items, prevItems)) {
+    setPrevItems(items);
+    if (!isEqual(items, getItemValues(draftItems))) {
+      setDraftItems(createItems(items));
     }
```

Comment:
I don't think you need this condition here, when `items` changes, you can just call `setDraftItems` and `setPrevItems`.

Reply:
This check needs to be there so that when local changes are made, the state isn't immediately overriden by the items prop. This wouldn't normally be a problem but in this case it cancels the sorting animation (the items just teleport instead of moving to their positions).